### PR TITLE
Fix GROUPING SETS of multiple empty sets

### PR DESCRIPTION
**Pull request by guofengrichard**
Currently if GROUPING SETS are empty sets, we will clean up
parse->groupClause and treat it as having no groups. If meanwhile there
are no aggregates in tlist or havingQual, the query would be planned as
if there are no GROUPING SETS, which is not correct.

This patch is just a workaround by petending the query has aggregates
when it finds the GROUPING SETS are empty sets.

Fix issue #11003

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/expected/gporca.out [hunk 1]
```diff
@@ -4322,27 +4322,7 @@ select 1 from orca.r group by ();
...
-        1
-        1
-        1
-        1
-        1
-        1
-        1
-        1
-(21 rows)
+(1 row)
```
Comment (asimrp) [inline src/test/regress/expected/gporca.out:4325]: Is this change intentional? If it is, then was the previous output of 21 rows wrong?

File: src/include/optimizer/planmain.h [hunk 2]
```diff
@@ -62,6 +62,8 @@ typedef struct GroupContext
 	double *p_dNumGroups;
 	List **pcurrent_pathkeys;
 	bool *querynode_changed;
+
+	bool	pretend_has_agg;
```
Comment (asimrp) [inline src/include/optimizer/planmain.h:66]: Would this change break any extensions?  In other words, is this an ABI breaking change?

File: src/backend/cdb/cdbgroup.c [hunk 3]
```diff
@@ -473,7 +473,7 @@ cdb_grouping_planner(PlannerInfo *root,
 	Plan	   *result_plan = NULL;
 	List	   *sub_tlist = NIL;
 	bool		has_groups = root->parse->groupClause != NIL;
-	bool		has_aggs = agg_costs->numAggs > 0;
+	bool		has_aggs = agg_costs->numAggs > 0 ? true : group_context->pretend_has_agg;
```
Comment (paul-guo-) [inline src/backend/cdb/cdbgroup.c:476]: `agg_costs->numAggs > 0 || group_context->pretend_has_agg;` seems to look better. Did not try to understand the code but better explain this new variable in the comment?

File: src/test/regress/expected/gporca.out (hunk 1 above)
Comment (guofengrichard) [reply src/test/regress/expected/gporca.out:4325]: Yes. We were using wrong output all the time before.

File: src/include/optimizer/planmain.h (hunk 2 above)
Comment (guofengrichard) [reply src/include/optimizer/planmain.h:66]: Hmm.. I don't think it changes the ABI.

File: src/backend/cdb/cdbgroup.c (hunk 3 above)
Comment (guofengrichard) [reply src/backend/cdb/cdbgroup.c]: > `agg_costs->numAggs > 0 || group_context->pretend_has_agg;` seems to look better. 

Yes, thanks.

**Review by asimrp — APPROVED** [review]
